### ISSUE TITLE
bug: cannot use `<C-j>` and `<C-k>` in fzf-lua to navigate the results

### ISSUE THREAD

**Issue by farzadmf** [LazyVim/LazyVim#3550]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.10.0

### Operating system/version

MacOS 14.4.1

### Describe the bug

I'm adding [fzf-lua](https://github.com/ibhagwan/fzf-lua) to my list of plugins.

I have assigned `<leader>/` to open the fzf-lua search for buffer lines.

If I don't enable `LazyVim`, in fzf-lua window, I can use `<C-j>` and `<C-k>` to navigate between the results (going down and up), but when I enable `LazyVim`, those keys no longer work (they close the fzf-lua window)

### Steps To Reproduce

1. Set up `neovim` as instructed by **Repro** section below.
2. Start `neovim`, open a file, and press `<leader>/` (which is `<space>/`)
3. Try to navigate the results using `<C-j>` and `<C-k>` and see if it doesn't work

Note:
If `lazyvim` is commented out (line `17`), in step `3` above, `<C-j>` and `<C-k>` would go down and up in the results.

### Expected Behavior

I'm not sure what setting in LazyVim is causing this issue. I thought it's the window movement mappings, but I tried to do:
```lua
vim.keymap.del('n', '<C-h>')
vim.keymap.del('n', '<C-j>')
vim.keymap.del('n', '<C-k>')
vim.keymap.del('n', '<C-l>')
```

But I think for `fzf-lua`, the `<C-j>` and `<C-k>` being used are somehow in `insert` mode, so disabling these won't affect those (and indeed it doesn't, but these are the only ones that came to my mind that might cause issues)

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./repro", ":p")

for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath })
end
vim.opt.runtimepath:prepend(lazypath)

vim.g.mapleader = " "

-- install plugins
local plugins = {
  { "LazyVim/LazyVim", import = "lazyvim.plugins" },
  {
    "ibhagwan/fzf-lua",
    url = "https://github.com/ibhagwan/fzf-lua",
    cmd = { "FzfLua" },
    dependencies = {
      "junegunn/fzf.vim",
    },
    keys = {
      { "<leader>/", "<cmd>FzfLua blines<cr>", desc = "buffer lines (fzf)" },
    },
    opts = {
      keymap = {
        builtin = {
          ["<C-f>"] = "preview-page-down",
          ["<C-b>"] = "preview-page-up",
        },
      },
    },
  },
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})
```

**Comment by folke** [comment]
It's the terminal mappings:

```lua
-- Terminal Mappings
map("t", "<esc><esc>", "<c-\\><c-n>", { desc = "Enter Normal Mode" })
map("t", "<C-h>", "<cmd>wincmd h<cr>", { desc = "Go to Left Window" })
map("t", "<C-j>", "<cmd>wincmd j<cr>", { desc = "Go to Lower Window" })
map("t", "<C-k>", "<cmd>wincmd k<cr>", { desc = "Go to Upper Window" })
map("t", "<C-l>", "<cmd>wincmd l<cr>", { desc = "Go to Right Window" })
map("t", "<C-/>", "<cmd>close<cr>", { desc = "Hide Terminal" })
map("t", "<c-_>", "<cmd>close<cr>", { desc = "which_key_ignore" })
```

fyi, I'm actually creating a fzf-lua extra. Will probably release it later today

**Comment by folke** [comment]
Add the below to your fzf-lua spec:

```lua
{
    "ibhagwan/fzf-lua",
    event = "VeryLazy",
    opts = {},
    keys = {
      { "<esc>", "<cmd>close<cr>", ft = "fzf", mode = "t", nowait = true },
      { "<c-j>", "<Down>", ft = "fzf", mode = "t", nowait = true },
      { "<c-k>", "<Up>", ft = "fzf", mode = "t", nowait = true },
   }
}
...

**Comment by farzadmf** [comment]
Thank you very much @folke for the quick answer.

> fyi, I'm actually creating a fzf-lua extra. Will probably release it later today

Oh, exciting; I guess you read my mind 😉; looking forward to it

**Comment by farzadmf** [comment]
@folke I just saw your addition of the `fzf-lua` extra. Thank you!

However, I see it disables Telescope. I agree that the two plugins are somehow doing the same thing, but I've been personally using them together (some things in `fzf-lua` and some things in `telescope`).

I saw a [similar comment](https://github.com/LazyVim/LazyVim/commit/aa059e0d496e570f898644d3f18bc344ca179d38) about `luasnip` and `nvim-snippets` about being able to somehow override the disabling.

Maybe in the snippets case, it might be less applicable, but since I've been using `fzf-lua` and `telescope` together, I was wondering if there's any way (or any plan to support) to override what's disabled

**Comment by folke** [comment]
Then just add telescope with enabled = true to your own config

**Comment by farzadmf** [comment]
@folke an update on this: I _think_ `<C-j>` and `<C-k>` are still not working

Minimal repro:
```lua
 -- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

vim.g.mapleader = " "

-- install plugins
local plugins = {
  { 'LazyVim/LazyVim', import = 'lazyvim.plugins' },
  { import = 'lazyvim.plugins.extras.editor.fzf' },
  {
    'nvim-telescope/telescope.nvim',
    enabled = true,
    keys = {
      { '<leader>/', false },
    },
  },
  {
    'ibhagwan/fzf-lua',
    url = 'https://github.com/ibhagwan/fzf-lua',
    cmd = { 'FzfLua' },
    keys = {
      { '<leader>/', '<cmd>FzfLua blines<cr>', desc = 'buffer lines (fzf)' },
    },
    opts = {
      keymap = {
        builtin = {
          ['<C-f>'] = 'preview-page-down',
          ['<C-b>'] = 'preview-page-up',
        },
      },
    },
  },
}

require("lazy").setup(plugins, {
  root = root .. "/plugins",
})
```

If I do "buffer lines" (using `<leader>/`), I was expecting `<C-j>` and `<C-k>` to work as expected, but that's not the case.

I tried disabling `lazyvim` altogether and having the keys directly in there like this:
```lua
local plugins = {
  -- { 'LazyVim/LazyVim', import = 'lazyvim.plugins' },
  -- { import = 'lazyvim.plugins.extras.editor.fzf' },
  {
    'nvim-telescope/telescope.nvim',
    enabled = true,
    keys = {
      { '<leader>/', false },
    },
  },
  {
    'ibhagwan/fzf-lua',
    url = 'https://github.com/ibhagwan/fzf-lua',
    cmd = { 'FzfLua' },
    keys = {
      { '<leader>/', '<cmd>FzfLua blines<cr>', desc = 'buffer lines (fzf)' },
      { "<c-j>", "<Down>", ft = "fzf", mode = "t", nowait = true },
      { "<c-k>", "<Up>", ft = "fzf", mode = "t", nowait = true },
    },
    opts = {
      keymap = {
        builtin = {
          ['<C-f>'] = 'preview-page-down',
          ['<C-b>'] = 'preview-page-up',
        },
      },
    },
  },
}
```

And that's still not working.

It only works if I remove the bindings altogether (and disable `lazyvim` and its `fzf` extra)

**Comment by folke** [comment]
IT all works perfectly fine using the fzf-lua extra.
If it doesn't work for you, then you messed something up.

**Comment by dpetka2001** [comment]
I've also switched fully to fzf-lua Extra since yesterday and can confirm that `<C-j>/<C-k>` work as they should on my end.

**Comment by farzadmf** [comment]
> If it doesn't work for you, then you messed something up.

I thought that would be the case; that's why I tried with the second minimal repro snippet above; I event shrunk it down to be like this:
```lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  { "LazyVim/LazyVim", import = "lazyvim.plugins" },
  { import = "lazyvim.plugins.extras.editor.fzf" },
}

require("lazy").setup(plugins, {
  root = root .. "/plugins",
})
```

And still doesn't work. I thought the fact that I'm using this minimal config means that there's nothing on my side interfering with anything 🤔

**Comment by dpetka2001** [comment]
Are you sure you don't have any terminal mappings that might interfere? I just tried the minimal file you mention and it also works there as well.

**Comment by farzadmf** [comment]
> Are you sure you don't have any terminal mappings that might interfere? I just tried the minimal file you mention and it also works there as well.

OMG, you guys are starting to make me feel like I'm crazy 😆

**Comment by farzadmf** [comment]
This spec works:
```lua
local plugins = {
  {
    "ibhagwan/fzf-lua",
  },
}
```

And when I add `lazyvim` (not even the extra) seems to break `<C-j>` and `<C-k>`
```lua
local plugins = {
  { "LazyVim/LazyVim", import = "lazyvim.plugins" },
  {
    "ibhagwan/fzf-lua",
  },
}
```

This one closes the `fzf` popup when I press `<C-j>` and `<C-k>`.

And this one:
```lua
local plugins = {
  { "LazyVim/LazyVim", import = "lazyvim.plugins" },
  { import = "lazyvim.plugins.extras.editor.fzf" },
  {
    "ibhagwan/fzf-lua",
  },
}
```

Disables them completely and nothing happens when I press them

**Comment by dpetka2001** [comment]
Here is a screencast of the minimal file you mention

[Screencast 2024-06-11 23:11:56.webm](https://github.com/LazyVim/LazyVim/assets/12776461/54a1e738-2b6a-4137-b118-b981d51dbb28)

**Comment by folke** [comment]
Can you make sure your lazy.nvim is 100% up to date?
Even in that repro, you need to open lazy and update (or delete .repro)

**Comment by farzadmf** [comment]
Man, this is SOOO weird. I tried to do a screen recording (and I did but it's about 34MB and I don't know how to do a smaller one with Mac 🙁 ).

But steps:
1. Delete `.repro` folder (to make sure we're clean)
2. Do `nv -u repro.lua` and let it do its thing
3. While `lazy.nvim`'s window is open, press `U` to udpate everything
4. Quit and re-open `nv -u repro.lua`
5. Press `f` to open the files dialog
6. `<C-j>`, `<C-k>`, `<Up>`, and `<Down>` are all no-ops and don't do anything

So, even if I managed to upload my screen recording, at the end, you would see `fzf-lua` window appearing there for a while, and that was me pressing the keys and seeing nothing happens
